### PR TITLE
TASK: refactor e2e testing setup

### PR DESCRIPTION
**Pull request by dimaip**
TODO:

- [x] migrate the rest of the tests
- [ ] don't flush all caches

### CODE REVIEW

**Comment by johannessteu** [comment]
What about the circle ci stuff btw?

**Comment by dimaip** [comment]
@johannessteu on my list for tomorrow, this took a bit longer then expected...

File: .circleci/config.yml [hunk 1]
```diff
@@ -53,15 +53,16 @@ jobs:
       - checkout
       - restore_cache:
           keys:
-            - full-app-cache-{{ checksum "Build/CircleCi/composer.json" }}
+            - full-app-cache-{{ checksum "Tests/IntegrationTests/TestDistribution/composer.json" }}-{{ checksum "Tests/IntegrationTests/TestDistribution/Configuration/Settings.yaml" }}
```
Comment (johannessteu) [inline .circleci/config.yml]: Why do we need the Settings.yaml here? The cache will only affect packages installed via composer and we overrite the settings.yaml with the one from the repo anyways, right?

Comment (dimaip) [reply .circleci/config.yml]: but what if we change the Settings.yaml e.g. as I did for Flow  fix?

Comment (johannessteu) [reply .circleci/config.yml]: Hmm, would need to test actually. But i thought the change should be there as it's stored in git. Or maybe we would have to change the order? first restore cache and then run the checkout command. Anyway - doesn't hurt at all to get the checkusm

**Pull request merged**